packages/dart/noports_core/lib/src/srvd/srvd_impl.dart

```diff
@@ -241,17 +253,25 @@ class SrvdImpl implements Srvd {
       verbose,
     );
 
-    logger
-        .info("Spawning socket connector isolate with parameters $parameters");
+    logger.info("Spawning socket connector isolate"
+        " with parameters $parameters");
 
-    unawaited(Isolate.spawn<ConnectorParams>(socketConnector, parameters));
+    Isolate spawned =
+        await Isolate.spawn<ConnectorParams>(socketConnector, parameters);
 
-    PortPair ports = await receivePort.first;
+    logger.info('Waiting for isolate to send its port pair info');
+    PortPair ports;
+    try {
+      ports =
+          await receivePort.first.timeout(Duration(milliseconds: timeoutMs));
+    } on TimeoutException catch (_) {
+      throw TimeoutException('Timed out after ${timeoutMs}ms');
+    }
 
     logger.info('Received ports $ports in main isolate'
         ' for session ${srvdSessionParams.sessionId}');
 
-    return ports;
+    return (ports, spawned);
```

Comment:
Returning the spawned isolate as well as the ports, so that the calling code can kill the isolate if it is not needed